src/wing/CTiglWingBuilder.cpp

```diff
@@ -253,28 +319,63 @@ CTiglWingBuilder::operator PNamedShape()
     return BuildShape();
 }
 // Set the name of each wing face
-void CTiglWingBuilder::SetFaceTraits (const std::string& wingUID, PNamedShape loft, bool hasBluntTE)
+void CTiglWingBuilder::SetFaceTraits (const std::vector<double>& guideCurveParams, const std::string& shapeUID, PNamedShape shape, bool hasBluntTE)
 {
-    unsigned int nSegments = _wing.GetSegmentCount();
+    auto params = guideCurveParams;
+    assert(std::is_sorted(std::begin(params), std::end(params)));
+
+    bool hasGuideCurves = params.size() > 0.;
+
+    size_t nFacesPerSegment = 2; // Without trailing edge
+    size_t idx_leading_edge = 1;
+    if (hasGuideCurves) {
+        if (!contains(params, -1.)) {
+            params.insert(params.begin(), -1.);
+        }
+
+        if (!contains(params, 1.)) {
+            params.push_back(1.);
+        }
+
+        // find leading edge curve
+        idx_leading_edge = find_index(params.cbegin(), params.cend(), [] (double val) {
+            return fabs(val) < 1e-3;
+        });
+
+        if (idx_leading_edge == params.size()) {
+            // no guide curve at leading edge
+            LOG(ERROR) << "No guide curve at leading edge defined";
+            return;
+        }
+        nFacesPerSegment = params.size() - 1;
+    }
 
     // designated names of the faces
-    std::vector<std::string> names(3);
-    names[0]="Bottom";
-    names[1]="Top";
-    names[2]="TrailingEdge";
+    std::vector<std::string> names;
+    for (size_t i = 0; i < idx_leading_edge; ++i) {
+        names.push_back("Bottom");
+    }
+
+    for (size_t i = idx_leading_edge; i < nFacesPerSegment; ++i) {
+        names.push_back("Top");
+    }
```

Comment:
Will this break in line 391 if `NO_EXPLICIT_TE_MODELING` is defined?

Reply:
Good question. We should probably simply remove the ifdef completely, as it is now untested old good. What do you think?

Reply:
Yes! Kick it out